examples/vault-auto-unseal/README.md

```diff
@@ -56,6 +55,49 @@ we strongly recommend deploying the Vault cluster into the private subnets of a
   `vault write /sys/license "text=$LICENSE_KEY_TEXT"`. If you don't do that, Vault
   will re-seal after 30 minutes and you won't be able to write or read secrets anymore.
 
+### Seal
+
+Vault servers boot sealed, which means it can access its storage, but can't decrypt it.
+So you can't really do anything apart from unsealing it or checking the server status.
+When you unseal, it recreates the master key, which isn't stored anywhere and is
+used to read the stored decryption key, which then can decrypt the data, and then
+you can start performing other operations on Vault.
+
+Since vault uses [Shamir's Secret Sharing][shamir], which splits the master key into
+pieces, running `vault operator unseal <unseal key>` adds piece by piece until there
+are enough parts to reconstruct the master key. This is done on different machines in the
+vault cluster for better security. Vault remains unsealed until it reboots or until
+someone manually reseals it.
+
+### Auto-unseal
+
+Vault Enterprise has a feature that allows automatic unsealing via Amazon KMS. It
+allows operators to delegate the unsealing process to AWS, which is useful for failure
+situations or creation of ephemeral clusters. This process uses an AWS KMS key as
+a [seal wrap][seal_wrap] mechanism: it encrypts and decrypts Vault's master key.
```

Comment:
> which is useful for failure situations or creation of ephemeral clusters.

Can you elaborate on these use cases? What specific problem(s) does auto-unseal solve?

Reply:
Also, can you make it more clear that AWS KMS is used to encrypt/decrypt the entire Master Key and replaces Shamir's Secret Sharing?